### ISSUE TITLE
Check Firewall Issue

### ISSUE THREAD

**Issue by FA-Bubba** [SamDel/ChromeCast-Desktop-Audio-Streamer#84]
Running version 2.9.0

After rebooting my Router, then clicking "Reset Settings", DAS only finds 16 of 17 Devices/Groups.  I tried Scan Again...  but the missing Device was not found.

I started two Groups and One Device that is not in a Group, after briefly turning Green, all turned Pink, and displayed "Check Firewall".   Nothing had been changed in the Firewall, so what would I need to check?

**Comment by SamDel** [comment]
When a device turns pink, "Check Firewall", it means:
- there has been communication (control messages) with the device
- when you start the stream a control message is send to tell the device to connect to DAS and load the stream
- when the device doesn't load the stream within 15 seconds the device box turns pink. 
It shouldn't take 15 seconds to load the stream, most likely something in the network is blocking connections from the device to DAS.

Things you can do:
- Check if the firewall on the desktop is not blocking incoming connections for 'ChromeCast.Desktop.AudioStreamer' (to be sure).
- Check the settings of the router. Maybe updates were installed when you rebooted the router.

**Comment by FA-Bubba** [comment]
Thanks,  The Mesh Controller had been updated.  DAS worked OK after I rolled the Controller back to the previous version.  (NOT a trivial task).

The Ubuquiti/UniFi support group  were unable to offer any suggestions of figuring out what caused the issue, and even said it might be an error in the update, but they were just guessing.

Evidently, something in the Mesh (UniFi) Controller software didn't like the DAS Device connections.

I welcome any suggestions on how to troubleshoot or otherwise determine what was happening with the new Controller software; until then I will not be doing any more Controller updates.

**Comment by SamDel** [comment]
Is this the user guide of your device: [https://dl.ubnt.com/guides/UniFi/UniFi_Controller_V5_UG.pdf](https://dl.ubnt.com/guides/UniFi/UniFi_Controller_V5_UG.pdf)?

Maybe they changed some (predefined) firewall rules, or settings, in the update. You can compare the firewall rules & settings before and after the update (page 33 in the manual).

**Comment by FA-Bubba** [comment]
WOW!  Thank you for finding this!!!

This is a User Guide for my Mesh Controller, and much more detailed than the one I have...

I will document the current Firewall rules & settings, and save as a reference for the next update in case DAS Devices get bumped-out again.

Thanks!

**Comment by FA-Bubba** [comment]
Another Firewall "experience"...

I upgraded to v3.0, and my Speaker Groups changed to the amber color, showing the message "Check Firewall".  I then clicked on individual speakers, and they all worked.  While casting to the individual speakers, duplicate Device Boxes appeared for all of the Speaker Groups.  Since my individual speakers have fixed IP addresses, and the Groups seem to have arbitrary addresses assigned by DAS, I unchecked "Automatically start last used devices and groups at startup", unloaded DAS, and restarted.

This time DAS was unable to cast to any speaker or device; all Device Boxes were Amber, but there were no 'Check Firewall' messages/labels.

I uninstalled v3.0 & reinstalled v2.9.0, with "Automatically start last used devices and groups at startup" NOT checked, and all is working OK now.  (Sorry, I failed to save any Log files, though).


![OptionsScreen-2020-07-35](https://user-images.githubusercontent.com/4895674/88460014-35ee8980-ce67-11ea-8a58-279e91799379.jpg)

**Comment by SamDel** [comment]
> While casting to the individual speakers, duplicate Device Boxes appeared for all of the Speaker Groups.

A new group-box is only added when a discovery message is received ("Discovered device: ..." in the log) from the group with a different ID (logged on the same line). I don't know what happened, maybe the groups decided to change their id's, or something odd on the network, or ... 
DAS doesn't assign IP addresses to groups, the devices in a group elect a leader themselves, the IP address of the leader is broadcasted.

The only thing changed in 3.0 is the 'Minimize to systray when closing' option. I don't think that has to do with it.

> ... but there were no 'Check Firewall' messages/labels.

That's not normal behaviour, I don't know what happened there.

Was the Mesh controller updated, or not, when this happened?

**Comment by FA-Bubba** [comment]
This was over a week after I discovered the Mesh Controller issue, and rolled it back to the previous version.

Everything has been working OK since I uninstalled DAS 3.0 and reinstalled DAV 2.9.0

I somehow think the issue may be related to the "Automatically start last used devices and groups at startup, which I am not using for now.

I will continue to monitor and post any new developments.

**Comment by FA-Bubba** [comment]
After over 1 year of stability, DAS started showing "Check Firewall" on multiple devices...
I checked Windows Firewall, and saw the following:


![CheckFirewall](https://user-images.githubusercontent.com/4895674/135854551-f7cf69e0-1862-4647-8126-d7908f9707d5.jpg)

Here's the Windows Update History:....

![Win10Updates](https://user-images.githubusercontent.com/4895674/135854593-8b2610d4-c811-4f34-9cfd-8a58cbadf93f.jpg)

But I am still getting Check Firewall on some devices...
What else can I try?
...and how can I prevent Windows from doing this again?

**Comment by SamDel** [comment]
When you start a stream:

1. DAS sends a message to the device to load the desktop stream.
2. The device makes a connection to DAS to load the stream.

When step 2 takes longer then 15 seconds DAS shows "Check Firewall". 

In your case it looks like the firewall is configured right, so I think there's another issue. Doesn't the stream load at all, or does it load when you wait long enough?

**Comment by FA-Bubba** [comment]
What I first noticed was all Devices were dropping off line, and making that audible multi-tone attempted connection sound, but no devices would play....

When the display (on Oct 2) ALL devices were showing "Check Firewall".  At the time, I believed the Firewall was properly configured, so I tried:..
 - Reset Settings
 - Close DAS and restart
 - Reboot PC
 - Reboot Gateway
 NONE of those actions resolved the Check Firewall Issue -- it would display on EVERY Device as soon as I loaded DAS...

So I checked the Firewall, and saw the first image I posted above (TWO entries for DAS, one with Private checked & Public unchecked; and one with Private unchecked and Public checked.  I tried to delete either one of the entries, but neither would delete, so I eventually got ALL of the DAS Firewall checkboxes checked, and then DAS started working OK. and has worked OK for the past two days...

Since I didn't have any screenShots of the Firewall setup prior to this, I don't know which boxes were checked before the Oct 2 Windows update, but whatever was checked, it was OK, because DAS has not had a "Check Firewall" issue here for over a year.

What is disturbing to me is how a Windows Update could disrupt the Firewall Settings....  In any event, I now know to check the Firewall settings if/when this happens again.

Will DAS display "Check Firewall" for any other reasons?

THANKS!

**Comment by FA-Bubba** [comment]
PS:  When this first occurred on Oct 2, DAS was set to play to two speaker Groups, for a total of 9 speakers from 1 PC with DAS, and to 2 more Speakers on a 2nd PC with DAS...

When ALL Devices dropped off line, during one of the Resets, I selected individual Speakers, instead of Groups, but that didn't help.

Now with the Firewall changes I made, I am back to casting via Speaker Groups, and it's been OK for the past 2 days.

**Comment by SamDel** [comment]
Can it be [this](https://github.com/SamDel/ChromeCast-Desktop-Audio-Streamer/issues/63#issuecomment-723456664) happened again:

> I have discovered that Windows updates routinely change the Local Network from Private to PUBLIC, and this kills a number of things, including DAS 

?

**Comment by FA-Bubba** [comment]
YES!... GREAT CATCH... THANK YOU & My apologies for forgetting to check that.

Odd that the same update did not change that setting on my other 3 PCs.

**Comment by FA-Bubba** [comment]
I'm periodically getting the same false alert mentioned on my Oct 18 post above:
![2021-12-14-DeviceBoxWarning](https://user-images.githubusercontent.com/4895674/146217014-0c2342bc-1ade-4d77-9885-006d83a1f755.jpg)

This is with version 3.8.4.0...  and a "Reset Settings" fixed it.

What else can I check besides Router Firewall and Network Privacy Status?

Thank you!

**Comment by SamDel** [comment]
If "Reset Settings" fixed it then something went wrong with the communication with the device.  I can't think of anything you can check, but sometimes there's an error in the log.
